File: BackEnd/CasottoIdS/src/main/java/it/unicam/cs/CasottoIdS/models/Utente.java
package it.unicam.cs.CasottoIdS.models;

import org.springframework.data.annotation.Id;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class Utente {

    public void setIdUtente(String idUtente) {
        this.idUtente = idUtente;
    }

    @Id
    public String idUtente;
    private String nome;
    private String cognome;
    private List<Notifica> notifiche;
    private Ruolo ruolo;
    private List<Evento> eventiAderiti;



    public Utente(String idUtente,String nome, String cognome, Ruolo ruolo) {
        this.idUtente=idUtente;
        this.nome = nome;
        this.cognome = cognome;
        this.notifiche = new ArrayList<>();
        this.ruolo = ruolo;
        this.eventiAderiti = new ArrayList<>();
    }

    public void addNotificaById(Notifica nuovaNotifica) {
        notifiche.add(nuovaNotifica);

    }

    public void addEventoById(Evento nuovoEvento) {
        eventiAderiti.add(nuovoEvento);
    }

    /**
     * recupera l'identificativo dell'utente
     * @return l'identificativo dell'utente
     * */
    public String getIdUtente() {
        return idUtente;
    }

    /**
     * recupera il nome di un determinato di un utente
     * @return il nome dell'utente
     * */
    public String getNome() {
        return nome;
    }

    /**
     * @param nome
     * imposta il nome per un utente
     * */
    public void setNome(String nome) {
        this.nome = nome;
    }

    /**
     * recupera il cognome di un utente
     * @return il cognome dell'utente
     * */
    public String getCognome() {
        return cognome;
    }

    /**
     * @param cognome
     * imposta il cognome per un utente
     * */
    public void setCognome(String cognome) {
        this.cognome = cognome;
    }

    /**
     * recupera le notifiche da una lista
     * @return la lista delle notifiche
     * */
    public List<Notifica> getNotifiche() {
        return notifiche;
    }
    /**
     * @param notifiche
     * permette di impostare una lista di notifiche da inviare
     * */
    public void setNotifiche(List<Notifica> notifiche) {
        this.notifiche = notifiche;
    }

    /**
     * recupera le info sul ruolo di un utente all'interno del sito
     * @return il ruolo dell'utente
     * */
    public Ruolo getRuolo() {
        return ruolo;
    }

    /**
     * @param ruolo
     * imposta un ruolo per un determinato utente all'interno del sito
     * */
    public void setRuolo(Ruolo ruolo) {
        this.ruolo = ruolo;
    }

    public List<Evento> getEventiAderiti() {
        return eventiAderiti;
    }

    public void setEventiAderiti(List<Evento> eventiAderiti) {
        this.eventiAderiti = eventiAderiti;
    }

    @Override
    public String toString() {
        return "Utente{" +
                "idUtente='" + idUtente + '\'' +
                ", nome='" + nome + '\'' +
                ", cognome='" + cognome + '\'' +
                ", notifiche=" + notifiche +
                ", ruolo=" + ruolo +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Utente utente = (Utente) o;
        return Objects.equals(idUtente, utente.idUtente) && Objects.equals(nome, utente.nome) && Objects.equals(cognome, utente.cognome) && Objects.equals(notifiche, utente.notifiche) && ruolo == utente.ruolo;
    }

}

File: FrontEnd/Casotto/lib/models/Utente.dart
import 'package:casotto/models/Evento.dart';

import 'Notifica.dart';
import 'Ruolo.dart';

class Utente {
  late String idUtente;
  late String _nome;
  late String _cognome;
  late List<Notifica> _notifiche;
  late Ruolo _ruolo;
  late List<Evento> eventiAderiti;

  Utente(String idUtente, String nome, String cognome, List<Notifica> notifiche,
      Ruolo ruolo, List<Evento> eventiAderiti) {
    this.idUtente = idUtente;
    this._nome = nome;
    this._cognome = cognome;
    this._notifiche = notifiche;
    this._ruolo = ruolo;
    this.eventiAderiti = eventiAderiti;
  }

  String getIdUtente() {
    return this.idUtente;
  }

  void setIdUtente(String idUtente) {
    this.idUtente = idUtente;
  }

  String getNome() {
    return this._nome;
  }

  void setNome(String nome) {
    this._nome = nome;
  }

  String getCognome() {
    return this._cognome;
  }

  void setCognome(String cognome) {
    this._cognome = cognome;
  }

  List<Notifica> getNotifiche() {
    return this._notifiche;
  }

  void setNotifiche(List<Notifica> notifiche) {
    this._notifiche = notifiche;
  }

  Ruolo getRuolo() {
    return this._ruolo;
  }

  String getRuoloString() {
    switch (this._ruolo) {
      case Ruolo.OSPITE:
        return "OSPITE";
      case Ruolo.REGISTRATO:
        return "REGISTRATO";
      case Ruolo.LOGGATO:
        return "LOGGATO";
      case Ruolo.GESTORE:
        return "GESTORE";
      case Ruolo.ADDETTO_SPIAGGIA:
        return "ADDETTO SPIAGGIA";
    }
  }

  void setRuolo(Ruolo ruolo) {
    this._ruolo = ruolo;
  }

  @override
  String toString() {
    return "id utente: " +
        this.idUtente +
        " nome " +
        this._nome +
        " cognome: " +
        this._cognome +
        " notifiche: " +
        this._notifiche.toString() +
        " ruolo_: " +
        this._ruolo.toString();
  }
}
